src/connectivity-api/connectivity-qt/connectivityqt/connectivity.cpp

```diff
@@ -387,6 +390,34 @@ void Connectivity::setHotspotEnabled(bool enabled)
 
     // TODO Remove this when SDK switch widget isn't broken
     d->m_propertyCache->setProperty("HotspotEnabled", enabled);
+
+    // HACK: For switching MediaTek /dev/wmtWifi adapter mode between AP/STA
+    //       when hotspot enabled state is changed.
+    try
+    {
+        const char* wmtWifi_dev = "/dev/wmtWifi";
```

Comment:
Please use QString here, or event better remove it all together and just pass it directory to the function using a QStringLitteral, Example like this
```
QFile wmtWifi_file(QStringLiteral("/dev/wmtWifi"));
if (wmtWifi_file.exists())....
```